Source/Engine/Core/Math/BoundingBox.cs

```diff
@@ -115,6 +115,57 @@ public BoundingBox(Vector3 minimum, Vector3 maximum)
             Maximum = maximum;
         }
 
+        /// <summary>
+        /// Check if this box is visible on some camera view.
+        /// </summary>
+        /// <param name="camera">The camera that observe this box.</param>
+        /// <returns></returns>
+        public bool IsOnView(Camera camera)
+        {
+            if (!camera)
+            {
+                return false;
+            }
+
+            Vector3 cameraPosition = camera.Position;
+            Vector3 cameraForward = camera.Transform.Forward;
+
+            // check if the camera is inside of box
+            if (CollisionsHelper.BoxContainsPoint(ref this, ref cameraPosition) == ContainmentType.Contains)
+            {
+                return true;
+            }
+
+            // check the box is on front of camera view
+            Ray ray = new Ray(cameraPosition, cameraForward);
+            if (CollisionsHelper.RayIntersectsBox(ref ray, ref this, out Vector3 point))
+            {
+                if (camera.IsPointOnView(point))
+                {
+                    return true;
+                }
+            }
+
+            // verify all box points to check if is on field of view
+            for (int i = 0; i < GetCorners().Length; i++)
+            {
+                // even if the corner isn't on screen return true if the box is on front of camera
+                // and on field on screen
+
+                float distanceOfCorner = Vector3.Distance(cameraPosition, GetCorners()[i]);
```

Comment:
Replace `float` with `Real` to get rid of CI errors from testing it with UseLargeWorlds

Reply:
@Withaust  I need use Float3 instead of Vector3?

Reply:
> @Withaust I need use Float3 instead of Vector3?

After looking at implementation of other methods in `BoundingBox`, `Vector3` should be fine.